domains/pallets/messenger/src/messages.rs

```diff
@@ -340,6 +349,16 @@ impl<T: Config> Pallet<T> {
         let req_msg = Outbox::<T>::take((dst_chain_id, channel_id, nonce))
             .ok_or(Error::<T>::MissingMessage)?;
 
+        OutboxMessageCount::<T>::try_mutate(
```

Comment:
Similar question here: should we remove the message and its weight tag, then decrement the message count?